CodeSniffer/Standards/Squiz/Sniffs/Commenting/FunctionCommentSniff.php

```diff
@@ -650,10 +650,12 @@ protected function processParams($commentStart, $commentEnd)
                     } else if (count($typeNames) === 1) {
                         // Check type hint for array and custom type.
                         $suggestedTypeHint = '';
-                        if (strpos($suggestedName, 'array') !== false) {
+                        if (strpos($suggestedName, 'array') !== false || substr($suggestedName, -2) === '[]') {
                             $suggestedTypeHint = 'array';
                         } else if (strpos($suggestedName, 'callable') !== false) {
                             $suggestedTypeHint = 'callable';
+                        } else if (strpos($suggestedName, 'callback') !== false) {
```

Comment:
It has never been a type, but has been allowed in @param tags. $suggestedName matches the type in the @param tag, $suggestedTypeHint matches the @param tag.